rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -857,7 +931,13 @@ namespace
 			{
 				if constexpr (std::is_same<T, u16>::value)
 				{
-					if (s_use_sse4_1)
+					if (s_use_avx2)
+					{
+						u32 iterations = length >> 4;
+						written = length & 0xF;
```

Comment:
These masks are wrong. We want `written = length - (length % 16)` while this only writes `written = (length % 16)`. This means we're actually using the fallback for all but the first N entries.

Reply:
P.S, Its just `length & ~0xF`